lib/src/browser/disposable_browser.dart

```diff
@@ -12,11 +12,191 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import 'dart:async';
 import 'dart:html';
 
+import 'package:meta/meta.dart';
+import 'package:w_common/func.dart';
+
 import 'package:w_common/src/common/disposable.dart' as disposable_common;
 
-class Disposable extends disposable_common.Disposable {
+class _InnerDisposable extends disposable_common.Disposable {
+  Func<Future<Null>> onDisposeHandler;
+
+  @override
+  Future<Null> onDispose() {
+    return onDisposeHandler();
+  }
+}
+
+/// Allows the creation of managed objects, including helpers for common
```

Comment:
I copy/pasted the doc comment from the VM version of `Disposable` here since I don't think it would otherwise be picked up by the API docs.